### PR TITLE
tools/dm: add troubleshooting.md

### PR DESCRIPTION
**Pull request by yikeke**
Via: https://github.com/pingcap/docs/blob/master/tools/dm/troubleshooting.md
PTAL. @lilin90 @csuzhangxc @GregoryIan 
PTAL again. @amyangfei

### CODE REVIEW

File: tools/dm/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+### 执行 `query-task` 或查看日志时出现 `Access denied for user 'root'@'172.31.43.27' (using password: YES)`
+
+在所有 DM 配置文件中，数据库相关的密码都必须使用经 dmctl 加密后的密文（若数据库密码为空，则无需加密）。有关如何使用 dmctl 加密明文密码，参见[使用 dmctl 加密上游 MySQL 用户密码](/tools/dm/deployment.md#使用-dmctl-加密上游-MySQL-用户密码)。
+
+此外，在 DM 运行过程中，上下游数据库的用户必须具备相应的读写权限。在启动同步任务过程中，DM 会自动进行相应权限的前置检查，详见[上游 MySQL 实例配置前置检查](/tools/dm/manage-task.md#上游-MySQL-实例配置前置检查)。
+
+### 处理不兼容的 DDL 语句
+
+你需要使用 dmctl 手动处理 TiDB 不兼容的 DDL 语句（包括手动跳过该 DDL 语句或使用用户指定的 DDL 语句替换原 DDL 语句，详见[跳过或替换异常 SQL 语句](#跳过或替换异常-SQL-语句)）。
```
Comment (csuzhangxc) [inline tools/dm/troubleshooting.md]: will *跳过或替换异常 SQL 语句* link to another file?

File: tools/dm/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+> **Note:** TiDB 目前并不兼容 MySQL 支持的所有 DDL 语句，详见 [TiDB 已支持的 DDL 语句](/sql/ddl.md)。
+
+## 重置数据同步任务
+
+在以下情况中，你需要重置整个数据同步任务：
+
+- 上游数据库中人为执行了 `RESET MASTER`，造成 relay log 同步出错
+- relay log 或上游 binlog 事件损坏或者丢失
```
Comment (csuzhangxc) [inline tools/dm/troubleshooting.md]: I think *binlog event* is better than *binlog 事件*.

**Comment by amyangfei** [comment]
> Why the `Skip or replace abnormal SQL statements` part in English doc is not translated, is this part moved to other page or some other reason?

well I find this part is outdated and we have a standalone page for `Skip or Replace Abnormal SQL Statements`. It is ok now

File: tools/dm/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    ```bash
+    resume-task ${task name}
+    ```
+
+但在某些情况下，你还需要重置数据同步任务。有关何时需要重置以及如何重置，详见[重置数据同步任务](#重置数据同步任务)。
+
+## 常见错误
+
+### 执行 `query-task` 或查看日志时出现 `Access denied for user 'root'@'172.31.43.27' (using password: YES)`
```
Comment (amyangfei) [inline tools/dm/troubleshooting.md]:
```suggestion
### 执行 `query-status` 或查看日志时出现 `Access denied for user 'root'@'172.31.43.27' (using password: YES)`
```

File: tools/dm/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+title: Data Migration 故障诊断
+category: tools
+---
+
+# Data Migration 故障诊断
+
+本文总结了在 DM 工具使用过程中一些常见的问题，并提供了相应的解决方案。
+
+如果你在运行 DM 工具时出现了错误，请尝试以下解决方案：
+
```
Comment (amyangfei) [inline tools/dm/troubleshooting.md]: we can add one more procedure here
1. 执行 `query-status` 命令查看任务运行状态和相关错误输出。

**Comment by GregoryIan** [comment]
LGTM

File: tools/dm/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+在所有 DM 配置文件中，数据库相关的密码都必须使用经 dmctl 加密后的密文（若数据库密码为空，则无需加密）。有关如何使用 dmctl 加密明文密码，参见[使用 dmctl 加密上游 MySQL 用户密码](/tools/dm/deployment.md#使用-dmctl-加密上游-MySQL-用户密码)。
+
+此外，在 DM 运行过程中，上下游数据库的用户必须具备相应的读写权限。在启动同步任务过程中，DM 会自动进行相应权限的前置检查，详见[上游 MySQL 实例配置前置检查](/tools/dm/manage-task.md#上游-MySQL-实例配置前置检查)。
+
+### 处理不兼容的 DDL 语句
+
+你需要使用 dmctl 手动处理 TiDB 不兼容的 DDL 语句（包括手动跳过该 DDL 语句或使用用户指定的 DDL 语句替换原 DDL 语句，详见[跳过或替换异常 SQL 语句](/tools/dm/skip-replace-sqls.md)）。
+
+> **Note:** TiDB 目前并不兼容 MySQL 支持的所有 DDL 语句，详见 [TiDB 已支持的 DDL 语句](/sql/ddl.md)。
```
Comment (lilin90) [inline tools/dm/troubleshooting.md]:
```suggestion
> **注意**：TiDB 目前并不兼容 MySQL 支持的所有 DDL 语句，详见 [TiDB 已支持的 DDL 语句](/sql/ddl.md)。
```

**Pull request merged**